redash/query_runner/snowflake.py

```diff
@@ -95,15 +125,25 @@ def _get_connection(self):
                 host = "{}.{}.snowflakecomputing.com".format(account, region)
             else:
                 host = "{}.snowflakecomputing.com".format(account)
-
-        connection = snowflake.connector.connect(
-            user=self.configuration["user"],
-            password=self.configuration["password"],
-            account=account,
-            region=region,
-            host=host,
-            application="Redash/{} (Snowflake)".format(__version__.split("-")[0]),
-        )
+                
+        if self.configuration.__contains__("password"):
+            connection = snowflake.connector.connect(
+                user=self.configuration["user"],
+                password=self.configuration["password"],
+                account=account,
+                region=region,
+                host=host,
+                application="Redash/{} (Snowflake)".format(__version__.split("-")[0]),
+            )
+        else:
+            connection = snowflake.connector.connect(
+                user=self.configuration["user"],
+                private_key=self._get_private_key(),
+                account=account,
+                region=region,
+                host=host,
+                application="Redash/{} (Snowflake)".format(__version__.split("-")[0]),
+            )
```

Comment:
Here's what @yoshiokatsuneo meant in terms of reducing duplication:

```suggestion
        params = {
                'user': self.configuration["user"],
                'account': account,
                'region': region,
                'host': host,
                'application': "Redash/{} (Snowflake)".format(__version__.split("-")[0]),
        }
        
        if 'password' in self.configuration:
            params['password'] = self.configuration['password']
        elif 'private_key' in self.configuration:
            params['private_key'] = self._get_private_key()
        else:
            raise Exception("Password or private key not set.")
        connection = snowflake.connector.connect(**params)
```

(I typed this in the GitHub UI, so you would want to double check it)

Reply:
Also added handling of the case when no password or private key is set.

Reply:
@arikfr 

Thank you very much for much clear comment !